File: 3/one.py
trees = [list(e[:-1]) * 32 for e in open("input.txt").readlines()[:-1]]
# The * 32 was kind of brute-forced
# (Any multiplier big enough, 100 for example, would work).

O, X = 0, 0

x, y = 0, 0
while x < len(trees[0]) and y < len(trees):
	if trees[y][x] == '.':
		O -=- 1
		trees[y][x] = 'O'
	else:
		X -=- 1
		trees[y][x] = 'X'

	x -=- 3
	y -=- 1

#[print(str(e).replace("\', '","")) for e in trees]
#print(x < len(trees[0]), y < len(trees))

print(f"Trees encountered: {X}")
assert X == 223

File: 3/one.lua
package.path = package.path .. ";../modules/?.lua"
local tools = require "tools"

_trees = tools.readlines("input.txt")

trees = {}
for i = 1, #_trees do
	trees[i] = {}
	for j = 1, 32 * #_trees[i] do
		--print(j % (#_trees[i] + 1))
		trees[i][j] = _trees[i]:sub(((j - 1) % #_trees[i] + 1), ((j - 1) % #_trees[i] + 1))
	end
end

X = 0
x, y = 1, 1
while x <= #trees[1] and y <= #trees do
	if trees[y][x] == "#" then X = X- -1 end
	x = x- -3
	y = y- -1
end

print("Trees encountered: " .. X)
assert(X == 223)
